core/src/main/java/io/questdb/std/Files.java

```diff
@@ -187,31 +185,31 @@ public static long length(LPSZ lpsz) {
 
     public static native int lock(long fd);
 
-    public static int mkdir(LPSZ path, int mode) {
+    public static int mkdir(Path path, int mode) {
         return mkdir(path.address(), mode);
     }
 
-    public static int mkdirs(LPSZ path, int mode) {
-        try (Path pp = new Path()) {
-            for (int i = 0, n = path.length(); i < n; i++) {
-                char c = path.charAt(i);
-                if (c == File.separatorChar) {
+    public static int mkdirs(Path path, int mode) {
+        for (int i = 0, n = path.length(); i < n; i++) {
+            char c = path.charAt(i);
+            if (c == File.separatorChar) {
 
-                    if (i == 2 && Os.type == Os.WINDOWS && path.charAt(1) == ':') {
-                        pp.put(c);
-                        continue;
-                    }
+                // do not attempt to create '/' on linux or 'C:\' on Windows
+                if ((i == 0 && Os.type != Os.WINDOWS) || (i == 2 && Os.type == Os.WINDOWS && path.charAt(1) == ':')) {
+                    continue;
+                }
 
-                    pp.$();
-                    if (pp.length() > 0 && !Files.exists(pp)) {
-                        int r = Files.mkdir(pp, mode);
-                        if (r != 0) {
-                            return r;
-                        }
+                // replace separator we just found with \0
+                // temporarily truncate path to the directory we need to create
+                path.$at(i);
+                if (path.length() > 0 && !Files.exists(path)) {
+                    int r = Files.mkdir(path, mode);
+                    if (r != 0) {
+                        path.put(i, File.separatorChar);
```

Comment:
nit/style: I would prefer try-finally instead of reverting to the original separator in 2 different places.

Reply:
Try…finally has an interesting code coverage artefact. It loses points if finally is not hit by exception. Which is the case here. So the repetition is intentional and C-like

Reply:
ok, I didn't know that. 

I guess the intent of the coverage tool was this: If you have code in the finally block then you are assuming an exceptional return is possible. If there is no exception observed then it means the exceptional code path is not tested -> less coverage points. Except this does not hold when you have multiple regular returns.